kafka-connect-neo4j/src/main/kotlin/streams/kafka/connect/common/ConfigurationMigrator.kt

```diff
@@ -0,0 +1,164 @@
+        RETRY_BACKOFF_MSECS to PropertyConverter("neo4j.max-retry-time") { convertMsecs(settings[RETRY_BACKOFF_MSECS] as String) },
+        RETRY_MAX_ATTEMPTS to PropertyConverter("neo4j.max-retry-attempts") {settings[RETRY_MAX_ATTEMPTS] as String},
+        // Sink
+        TOPIC_CDC_SOURCE_ID to PropertyConverter("neo4j.cdc.source-id.topics") {settings[TOPIC_CDC_SOURCE_ID] as String},
+        TOPIC_CDC_SOURCE_ID_LABEL_NAME to PropertyConverter("neo4j.cdc.source-id.label-name") {settings[TOPIC_CDC_SOURCE_ID_LABEL_NAME] as String},
+        TOPIC_CDC_SOURCE_ID_ID_NAME to PropertyConverter("neo4j.cdc.source-id.property-name") {settings[TOPIC_CDC_SOURCE_ID_ID_NAME] as String},
+        TOPIC_CDC_SCHEMA to PropertyConverter("neo4j.cdc.schema.topics") {settings[TOPIC_CDC_SCHEMA] as String},
+        BATCH_PARALLELIZE to PropertyConverter("") {settings[BATCH_PARALLELIZE] as String},
+        TOPIC_CUD to PropertyConverter("neo4j.cud.topics") {settings[TOPIC_CUD] as String},
+        TOPIC_PATTERN_MERGE_NODE_PROPERTIES_ENABLED to PropertyConverter("neo4j.pattern.node.merge-properties") {settings[TOPIC_PATTERN_MERGE_NODE_PROPERTIES_ENABLED] as String},
+        TOPIC_PATTERN_MERGE_RELATIONSHIP_PROPERTIES_ENABLED to PropertyConverter("neo4j.pattern.relationship.merge-properties") {settings[TOPIC_PATTERN_MERGE_RELATIONSHIP_PROPERTIES_ENABLED] as String},
+        // Source
+        BATCH_TIMEOUT_MSECS to PropertyConverter("neo4j.batch-timeout") { convertMsecs(settings[BATCH_TIMEOUT_MSECS] as String) },
+        TOPIC to PropertyConverter("neo4j.query.topic") {settings[TOPIC] as String},
+        STREAMING_FROM to PropertyConverter("") {settings[STREAMING_FROM] as String},
+        SOURCE_TYPE to PropertyConverter("") {settings[SOURCE_TYPE] as String},
+        SOURCE_TYPE_QUERY to PropertyConverter("neo4j.query") {settings[SOURCE_TYPE_QUERY] as String},
+        STREAMING_PROPERTY to PropertyConverter("neo4j.query.streaming-property") {settings[STREAMING_PROPERTY] as String},
+        STREAMING_POLL_INTERVAL to PropertyConverter("neo4j.query.poll-interval") { convertMsecs(settings[STREAMING_POLL_INTERVAL] as String) },
+        ENFORCE_SCHEMA to PropertyConverter("") {settings[ENFORCE_SCHEMA] as String}
+    )
+
+    // Configuration properties that have user-defined keys
+    private val prefixConverterMap: Map<String, String> = mutableMapOf(
+        Neo4jSinkConnectorConfig.TOPIC_PATTERN_NODE_PREFIX to "neo4j.pattern.node.topic.",
+        Neo4jSinkConnectorConfig.TOPIC_PATTERN_RELATIONSHIP_PREFIX to "neo4j.pattern.relationship.topic.",
+        Neo4jSinkConnectorConfig.TOPIC_CYPHER_PREFIX to "neo4j.cypher.topic."
+    )
+
+    /**
+     * Migrate configuration keys from existing format to v5.1 connector format.
+     * Configuration properties containing msecs units are converted to new format.
+     *
+     * @return updated configuration key-value pairs
+     */
+    fun migrate(): Map<String, Any> {
```

Comment:
nit: I'd go as far as explicitly naming this "migrateToV51"

Reply:
Is there a clearer way to represent 5.1 in camelcase?
V51 to me looks like it could be version 51.
e.g. 5_1, 5dot1?

Reply:
I'll let you pick and/or let others chime in, I have no strong preference here